test/integration/controller/jobs/pod/pod_controller_test.go

```diff
@@ -0,0 +1,402 @@
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/controller/jobs/pod"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	podtesting "sigs.k8s.io/kueue/pkg/util/testingjobs/pod"
+	"sigs.k8s.io/kueue/pkg/workload"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+const (
+	podName = "test-pod"
+)
+
+var _ = ginkgo.Describe("Pod controller", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+	ginkgo.When("manageJobsWithoutQueueName is disabled", func() {
+		var defaultFlavor = testing.MakeResourceFlavor("default").Label("kubernetes.io/arch", "arm64").Obj()
+
+		ginkgo.BeforeAll(func() {
+			fwk = &framework.Framework{
+				CRDPath:     crdPath,
+				WebhookPath: webhookPath,
+			}
+			cfg = fwk.Init()
+			ctx, k8sClient = fwk.RunManager(cfg, managerSetup(
+				jobframework.WithManageJobsWithoutQueueName(false),
+				jobframework.WithKubeServerVersion(serverVersionFetcher),
+				jobframework.WithPodSelector(&metav1.LabelSelector{}),
+				jobframework.WithPodNamespaceSelector(&metav1.LabelSelector{
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key:      "kubernetes.io/metadata.name",
+							Operator: metav1.LabelSelectorOpNotIn,
+							Values:   []string{"kube-system", "kueue-system"},
+						},
+					},
+				}),
+			))
+			gomega.Expect(k8sClient.Create(ctx, defaultFlavor)).To(gomega.Succeed())
+		})
+		ginkgo.AfterAll(func() {
+			util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, defaultFlavor, true)
+			fwk.Teardown()
+		})
+
+		var (
+			ns          *corev1.Namespace
+			wlLookupKey types.NamespacedName
+		)
+
+		ginkgo.BeforeEach(func() {
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "pod-namespace-",
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+			wlLookupKey = types.NamespacedName{Name: pod.GetWorkloadNameForPod(podName), Namespace: ns.Name}
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should reconcile the single pod with the queue name", func() {
+			pod := podtesting.MakePod(podName, ns.Name).Queue("test-queue").Obj()
+			gomega.Expect(k8sClient.Create(ctx, pod)).Should(gomega.Succeed())
+			lookupKey := types.NamespacedName{Name: podName, Namespace: ns.Name}
+			createdPod := &corev1.Pod{}
+			gomega.Eventually(func() error {
+				return k8sClient.Get(ctx, lookupKey, createdPod)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			gomega.Expect(hasKueueSchedulingGate(createdPod)).To(gomega.BeTrue(),
+				"Pod should have 'kueue.x-k8s.io/admission' scheduling gate set")
+
+			gomega.Expect(hasManagedLabel(createdPod)).To(gomega.BeTrue(),
+				"Pod should have 'kueue.x-k8s.io/managed=true' label set")
+
+			gomega.Expect(createdPod.Finalizers).To(gomega.ContainElement("kueue.x-k8s.io/managed"),
+				"Pod should have 'kueue.x-k8s.io/managed' finalizer set")
```

Comment:
```suggestion
			gomega.Expect(createdPod.Finalizers).To(gomega.ContainElement("kueue.x-k8s.io/managed"),
				"Pod should have finalizer")
```

The rest is redundant (it will be visible in the error message).